tests/webservice_test.py

```diff
@@ -270,39 +271,20 @@ def test_unknown_error(self):
             self.run_client(self.client.country(ip))
 
     def test_request(self):
-
-        request = None
-
-        def custom_handler(r):
-            nonlocal request
-            request = r
-            return ""
-
+        matcher = HeaderValueMatcher({
+            "Accept": "application/json",
+            "Authorization":"Basic NDI6YWJjZGVmMTIzNDU2",                
+            "User-Agent": lambda x: x.startswith("GeoIP2-Python-Client/"),})
```

Comment:
I don't think this is the correct usage of `HeaderValueMatcher`. I think you still set `headers`, but you would want to define a custom matcher for "User-Agent".

[See the docs](https://pytest-httpserver.readthedocs.io/en/latest/howto.html#advanced-header-matching).